### DISCUSSION TITLE
Why does GitHub change the data format from LF to CRLF without warning the user?

### DISCUSSION THREAD

**Discussion by AberQ** [community/community#144331] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

When saving my Django project, github changed the type of my docker files from LF to CRLF without my knowledge and because of this, the project did not start

**Comment by manqur1** [comment]
GitHub doesn’t automatically change line endings from LF to CRLF, but this can happen when different environments are configured to handle line endings differently, especially if you're working on a project across Windows (which defaults to CRLF) and Unix-based systems (which use LF). Here’s what’s likely happening and how to prevent it:

Why It Happens:
1. Git Configuration: By default, Git is configured to handle line endings in a way that works across different platforms. The setting core.autocrlf is key here:

- core.autocrlf = true (default on Windows): Git converts LF to CRLF when checking out files on Windows and converts CRLF back to LF when committing.
- core.autocrlf = input: Git keeps files with LF endings on check-out and on commits (suitable for Unix-based systems).
- core.autocrlf = false: Disables any automatic line-ending conversions.

If you or your collaborators are using different settings, Git might change the line endings during commits or checkouts.

2. Editor Settings: Some text editors or IDEs on Windows, like Notepad or even some settings in VSCode, can automatically save files with CRLF line endings, which Git might then commit if it doesn’t detect it.


To avoid this issue, set up a .gitattributes file in your project root to enforce line endings consistently across different environments:

```
# .gitattributes
* text=auto
*.sh text eol=lf
*.yml text eol=lf
*.dockerfile text eol=lf
*.py text eol=lf
*.html text eol=lf
```
* text=auto tells Git to handle line endings automatically.
* Specifying eol=lf for specific file types (e.g., .dockerfile, .py) ensures they always use LF, even on Windows.

**Reply by AberQ** [reply]
Thank you^_^

**Reply by manqur1** [reply]
Your welcome, Good Look
